next/frontend/hooks/useServerSideAuth.ts

```diff
@@ -0,0 +1,19 @@
+import { ServerSideAuthContext } from 'components/logic/ServerSideAuthProvider'
+import { Tier } from 'frontend/dtos/accountDto'
+import { useContext } from 'react'
+
+export const useServerSideAuth = () => useContext(ServerSideAuthContext)
```

Comment:
I suggest to refactor this to:
```
export const useServerSideAuth = () => {
  const context = useContext(ServerSideAuthContext);
  const tier = userData?.['custom:tier']

  return {
    ...context,
    isAuthenticated: !!userData,
    isLegalEntity: userData?.['custom:account_type'] === 'po',
    tierStatus: {
      tier,
      isIdentityVerified: tier === Tier.IDENTITY_CARD || tier === Tier.EID,
      isIdentityVerificationNotYetAttempted: !tier || tier === Tier.NEW,
    },
  }
}
```

Gets rid of cumbersome `useDerivedServerSideAuthState` name, no drawbacks.

Reply:
imo drawback is loosing the understanding between which data you get from the server and which data are computed on frontend from existing ones

i.e. you could check if user `isAuthenticated` against the auth lib server side, which is a stronger 'truth' than checking if userData exists and assuming authentication based on that. In the debugging process I'd like to know what came from the amplify lib unchanged and what is computed by us.

`useDerivedServerSideAuthState` is essentially just a bunch of helpers transforming the data, I like to keep these separate from app 'state' - if we weren't in react it might have been a regular (pure) function instead of a hook.

But - of course, no right way, this is just preference. I'm persuadable to the single-hook approach, it was how I had it originally, above is the reasoning for splitting it in two to be considered beforehand.